components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -777,6 +777,38 @@ public APIDefinitionValidationResponse validateAPIDefinition(String apiDefinitio
                 title = info.getTitle();
                 context = info.getTitle().replaceAll("\\s", "").toLowerCase();
             }
+
+            String xThrottlingLimitExtension = APIConstants.SWAGGER_X_THROTTLING_LIMIT;
+            if (openAPI.getExtensions() != null && openAPI.getExtensions().containsKey(xThrottlingLimitExtension)) {
+                ErrorItem errorItem = OASParserUtil.validateAPIThrottleLimitExtension(
+                        openAPI.getExtensions().get(xThrottlingLimitExtension));
+                if (errorItem != null) {
+                    validationResponse.getErrorItems().add(errorItem);
+                    errorItem.setDescription("Error while validating the x-throttling-limit extension " +
+                            "at the API Level");
+                    return validationResponse;
+                }
+            }
+            // Validate the x-throttling-limit extension for each resource
+            for (String path : openAPI.getPaths().keySet()) {
+                PathItem pathItem = openAPI.getPaths().get(path);
+                for (Map.Entry<PathItem.HttpMethod, Operation> operationEntry :
+                        pathItem.readOperationsMap().entrySet()) {
+                    Operation operation = operationEntry.getValue();
+                    if (operation.getExtensions() != null
+                            && operation.getExtensions().containsKey(xThrottlingLimitExtension)) {
+                        ErrorItem errorItem = OASParserUtil.validateAPIThrottleLimitExtension(
+                                operation.getExtensions().get(xThrottlingLimitExtension));
+                        if (errorItem != null) {
+                            validationResponse.getErrorItems().add(errorItem);
+                            errorItem.setDescription(String.format("Error while validating the x-throttling-limit " +
```

Comment:
This is basically we found a user issue right? Its not really an error during validation rather we located some issues after doing the validation
```suggestion
                            errorItem.setDescription(String.format("Invalid x-throttling-limit " +
```